### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by jsedge**

### CODE REVIEW

File: ene/entities.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        self.title = title
+        self.show_id = show_id
+        self.list_id = list_id
+        if episodes is None:
+            self.episodes = dict()
+        else:
+            self.episodes = episodes
+        self.key = key
+
+    def add_episode(self, episode):
```
Comment (MaT1g3R) [inline ene/entities.py]: I'm not sure what this method does

File: ene/series_manager.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+class SeriesManager:
+    """
+    Manages access to series
+    """
+    def __init__(self, config):
+        self._series = ShowList()
+        self._db = None
+        self._config = config
+
+    def init_db(self, data_home):
```
Comment (MaT1g3R) [inline ene/series_manager.py]: Should inject an instance `ShowDataAccess` instead (remove this method)

File: ene/persistence/data_access.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
+""" This module handles persistence of data to the database """
+from .models import ShowModel, EpisodeModel, EneDatabase
+
+
+class ShowDataAccess:
+    """
+    Contains methods to access and persist data to the SQLite database
+    """
+    def __init__(self):
```
Comment (MaT1g3R) [inline ene/persistence/data_access.py]: Inject EneDatabase instance instead

File: ene/series_manager.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+    def fetch_shows_from_db(self):
+        """
+        Fetches all shows from the database and adds them to the series list
+        """
+
+        for res in self._db.get_all_shows():
+            self._series.add(res)
+
+    def fetch_shows_from_files(self):
```
Comment (MaT1g3R) [inline ene/series_manager.py]: why are we making a new FileManager for all of these methods?

File: ene/entities.py [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    """
+    def __init__(self, title, show_id=None, list_id=None, episodes: dict = None, key=None):
+        self.title = title
+        self.show_id = show_id
+        self.list_id = list_id
+        if episodes is None:
+            self.episodes = dict()
+        else:
+            self.episodes = episodes
+        self.key = key
```
[the remark this answers is not in the document]
Comment (jsedge) [reply ene/entities.py]: Removed.

File: ene/entities.py [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+        NEW = 1
+        UNWATCHED = 2
+        WATCHED = 3
+
+    def __init__(self, path: Path, state=State.NEW, number=0, key=None):
+        self.path = path
+        self.state = state
+        self.name = path.name
+        self.number = number
+        self.key = key
```
[the remark this answers is not in the document]
Comment (jsedge) [reply ene/entities.py]: Renamed as episode_id as this is what it actually is

File: ene/entities.py [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+
+        temp = self.name.replace(title, '')
+        # Searches for a number that is not preceded by x or succeeded by x or p
+        num = re.search(r'(?<![\(x0-9])[0-9]+(?![0-9xp])', temp)
+        try:
+            if num is not None:
+                num = num[0]
+                self.number = int(num)
+            else:
+                self.number = -1
```
[the remark this answers is not in the document]
Comment (jsedge) [reply ene/entities.py]: Changed to None

File: ene/entities.py [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+        return self.number > other.number
+
+    def __eq__(self, other):
+        return self.path == other.path
+
+    def __hash__(self):
+        return hash(self.path)
+
+
+class ShowList(dict):
```
[the remark this answers is not in the document]
Comment (jsedge) [reply ene/entities.py]: Switched to UserDict, only overriding __missing__ so this should be fine

File: ene/series_manager.py (hunk 2 above)
Comment (jsedge) [reply ene/series_manager.py]: As per discussion this will be resolved later

**Pull request merged**